src/dashboard/src/components/administration/forms.py

```diff
@@ -129,10 +140,277 @@ class AtomDipUploadSettingsForm(SettingsForm):
+                        choices.append((item.pk, item.description))
+                elif ftype == 'storage_service':
+                    for loc in get_storage_locations(purpose=field['purpose']):
+                        choices.append((loc['resource_uri'], loc['description']))
+                self.fields[choice_uuid] = forms.ChoiceField(**opts)
+
+    def load_config(self, name):
+        """
+        Bound the choices found in the XML document to the form fields.
+        """
+        self.fields['name'].initial = name
+        self.fields['name'].widget.attrs['readonly'] = 'readonly'
+        config_path = os.path.join(helpers.processing_config_path(), '{}ProcessingMCP.xml'.format(name))
+        root = etree.parse(config_path)
+        for choice in root.findall('.//preconfiguredChoice'):
+            applies_to = choice.findtext('appliesTo')
+            go_to_chain = choice.findtext('goToChain')
+            fprops = self.processing_fields.get(applies_to)
+            field = self.fields.get(applies_to)
+            if fprops is None or go_to_chain is None or field is None:
+                continue
+            if fprops['type'] == 'days':
+                field.initial = int(float(choice.findtext('delay'))) / (24 * 60 * 60)
+            else:
+                field.initial = go_to_chain
+
+    def save_config(self):
+        """
+        Encode the configuration to XML and write it to disk.
+        """
+        name = self.cleaned_data['name']
+        del self.cleaned_data['name']
+        config_path = os.path.join(helpers.processing_config_path(), '{}ProcessingMCP.xml'.format(name))
+        config = PreconfiguredChoices(config_path)
+        for choice_uuid, value in self.cleaned_data.iteritems():
+            fprops = self.processing_fields.get(choice_uuid)
+            if fprops is None or value is None:
+                continue
+            field = self.fields.get(choice_uuid)
+            if field is None:
+                continue
+            if isinstance(field, forms.ChoiceField):
+                if not value: # Ignore empty string!
+                    continue
+            if fprops['type'] == 'days':
+                if value == 0:
+                    continue
+                delay = str(float(value) * (24 * 60 * 60))
+                config.add_choice(choice_uuid, fprops['chain'], delay_duration=delay, comment=fprops['label'])
+            else:
+                config.add_choice(choice_uuid, value, comment=fprops['label'])
+        config.save()
+
+
+def get_storage_locations(purpose):
+    try:
+        dirs = storage_service.get_location(purpose=purpose)
+        if len(dirs) == 0:
+            raise Exception('Storage server improperly configured.')
+    except Exception:
+        dirs = []
+    return dirs
+
+
+class PreconfiguredChoices(object):
+    """
+    Encode processing configuration XML documents and optionally write to disk.
+    """
+    def __init__(self, config_path):
```

Comment:
Nitpick, but since this is only used by `save`, maybe defer asking for a `config_path` until the document actually gets saved?